Tests/ArcGISToolkitTests/FloorFilterViewControllerTests.swift

```diff
@@ -0,0 +1,93 @@
+
+import XCTest
+@testable import ArcGISToolkit
+import ArcGIS
+
+class FloorFilterViewControllerTests: XCTestCase {
+    
+    func testTopPlacement() {
+        let floorFilterView = FloorFilterViewController.makeFloorFilterView(geoView: AGSGeoView(frame: .zero), xMargin: UIScreen.main.bounds.width - 100, yMargin: 100)
+        let xPositionOfFloorFilterView = UIScreen.main.bounds.width - CGFloat(100) - CGFloat(floorFilterView?.view.bounds.width ?? 0)
+        XCTAssertEqual(UIScreen.main.bounds.width - 100 - 50, xPositionOfFloorFilterView)
+    }
+    
+    func testBottomPlacement() {
+        let floorFilterView = FloorFilterViewController.makeFloorFilterView(geoView: AGSGeoView(frame: .zero), xMargin: 40, yMargin: UIScreen.main.bounds.height - 300)
+        let yPositionOfFloorFilterView = UIScreen.main.bounds.height - CGFloat(300) - CGFloat(floorFilterView?.view.bounds.height ?? 0)
+        let expectedHeight = ((50*2)+(50*3))
+        XCTAssertEqual(UIScreen.main.bounds.height - 300 - CGFloat(expectedHeight), yPositionOfFloorFilterView)
+    }
+    
+    func testSitesData() throws {
+        let portal = AGSPortal(url: URL(string: "https://indoors.maps.arcgis.com/")!, loginRequired: false)
+        let portalItem = AGSPortalItem(portal: portal, itemID: "f133a698536f44c8884ad81f80b6cfc7")
+        let map = AGSMap(item: portalItem)
+        let mapView = AGSMapView()
+        mapView.map = map
+        
+        XCTLoad(map)
+        let floorManager = try XCTUnwrap(map.floorManager)
+        let viewModel = FloorFilterViewModel()
+        viewModel.floorManager = floorManager
+        XCTAssertEqual(getSitesData().count, viewModel.sites.count)
+    
+    }
+    
+    func testFacilitiesData() throws {
+        let portal = AGSPortal(url: URL(string: "https://indoors.maps.arcgis.com/")!, loginRequired: false)
+        let portalItem = AGSPortalItem(portal: portal, itemID: "f133a698536f44c8884ad81f80b6cfc7")
+        let map = AGSMap(item: portalItem)
+        let mapView = AGSMapView()
+        mapView.map = map
+        
+        XCTLoad(map)
+        let floorManager = try XCTUnwrap(map.floorManager)
+        let viewModel = FloorFilterViewModel()
+        viewModel.floorManager = floorManager
+        viewModel.selectedSite = viewModel.sites.first
+        XCTAssertEqual(getFacilitiesData().count, viewModel.facilities.count)
+        
+    }
+    
+    func testLevelsData() {
+        let portal = AGSPortal(url: URL(string: "https://indoors.maps.arcgis.com/")!, loginRequired: false)
+        let portalItem = AGSPortalItem(portal: portal, itemID: "f133a698536f44c8884ad81f80b6cfc7")
+        let map = AGSMap(item: portalItem)
+        let mapView = AGSMapView()
+        mapView.map = map
+
+        XCTLoad(map)
+        let floorManager = try XCTUnwrap(map.floorManager)
+        let viewModel = FloorFilterViewModel()
+        viewModel.floorManager = floorManager
+        XCTAssertEqual(getLevelsData().count, viewModel.allLevels.count)
```

Comment:
If you're just comparing counts, you can just use `3`, as that value will never change and you then won't need to create the array.
```suggestion
        XCTAssertEqual(viewModel.allLevels.count, 3)
```